backend/de.metas.banking/de.metas.banking.camt53/src/main/java/de/metas/banking/camt53/wrapper/NoBatchReportEntry2Wrapper.java

```diff
@@ -0,0 +1,267 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.banking.camt53.wrapper;
+
+import ch.qos.logback.classic.Level;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import de.metas.adempiere.model.I_C_Invoice;
+import de.metas.banking.BankStatementId;
+import de.metas.banking.camt53.jaxb.camt053_001_02.ActiveOrHistoricCurrencyAndAmount;
+import de.metas.banking.camt53.jaxb.camt053_001_02.AmountAndCurrencyExchange3;
+import de.metas.banking.camt53.jaxb.camt053_001_02.AmountAndCurrencyExchangeDetails3;
+import de.metas.banking.camt53.jaxb.camt053_001_02.CurrencyExchange5;
+import de.metas.banking.camt53.jaxb.camt053_001_02.DateAndDateTimeChoice;
+import de.metas.banking.camt53.jaxb.camt053_001_02.EntryDetails1;
+import de.metas.banking.camt53.jaxb.camt053_001_02.EntryTransaction2;
+import de.metas.banking.camt53.jaxb.camt053_001_02.RemittanceInformation5;
+import de.metas.banking.camt53.jaxb.camt053_001_02.ReportEntry2;
+import de.metas.banking.camt53.jaxb.camt053_001_02.TransactionInterest2;
+import de.metas.banking.service.BankStatementLineCreateRequest;
+import de.metas.bpartner.BPartnerId;
+import de.metas.currency.CurrencyCode;
+import de.metas.currency.CurrencyRepository;
+import de.metas.document.engine.DocStatus;
+import de.metas.invoice.InvoiceId;
+import de.metas.invoice.service.IInvoiceDAO;
+import de.metas.logging.LogManager;
+import de.metas.money.CurrencyId;
+import de.metas.money.Money;
+import de.metas.organization.IOrgDAO;
+import de.metas.organization.OrgId;
+import de.metas.util.Check;
+import de.metas.util.Loggables;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.compiere.util.TimeUtil;
+import org.slf4j.Logger;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static de.metas.banking.camt53.jaxb.camt053_001_02.CreditDebitCode.CRDT;
+
+@Value
+public class NoBatchReportEntry2Wrapper
+{
+	private static final Logger logger = LogManager.getLogger(NoBatchReportEntry2Wrapper.class);
+
+	@NonNull
+	IOrgDAO orgDAO;
+
+	@NonNull
+	IInvoiceDAO invoiceDAO;
+
+	@NonNull
+	ReportEntry2 entry;
+
+	@NonNull
+	CurrencyRepository currencyRepository;
+
+	@Builder
+	private NoBatchReportEntry2Wrapper(
+			@NonNull final IOrgDAO orgDAO,
+			@NonNull final IInvoiceDAO invoiceDAO,
+			@NonNull final ReportEntry2 entry,
+			@NonNull final CurrencyRepository currencyRepository)
```

Comment:
minor, but pls re-order the params,
* have the services first
* have the data afterwards (i.e. `entry` param)